htdocs/core/class/html.formfile.class.php

```diff
@@ -1937,4 +1938,346 @@ public function showPreview($file, $modulepart, $relativepath, $ruleforpicto = 0
 		}
 		return $out;
 	}
+
+	/**
+	 * Return the HTML code for a small document view for the given module (typical used on the card tab of a module)
+	 *
+	 * @param	string			$module		The name of the module
+	 * @param	CommonObject	$object		The currrent used object of the module
+	 * @param	string			$lang		(optional) The language for the small document view
+	 * @return	string						The full HTML code for the small document view
+	 */
+	public function getDocumentsViewSmall($module, CommonObject $object, $lang = null)
+	{
+		global $conf, $user;
+
+		// security checks
+		$genallowed = $this->_canUserGenerateDocuments($user, $module, $object);
+		$delallowed = $this->_canUserDeleteDocuments($user, $module, $object);
+
+		// same/easy values for the most types of modules
+		$filename			= dol_sanitizeFileName($object->ref);
+		$subdir				= $filename;
+		$filedir			= $this->_getDocumentRootPath($module, $object)."/".$subdir;
+		$urlsource			= $_SERVER['PHP_SELF']."?id=".$object->id;
+		$modelpdf			= $object->modelpdf;
+		$allowgenifempty	= 1;
+		$param				= "";
+		$lang				= $lang == null ? $object->default_lang : $lang;
+
+		// TODO: will be 'remove_file_comfirm', when all module cards have a confirm delete (Sekan T., 2020-04-01)
+		$deleteaction		= 'remove_file';
+
+		// different/complex values for a hand of modules
+		switch($module)
+		{
+			case "actions":
```

Comment:
This are variables that depends on the page that need to show the documents. So this variable initialization should be done on the calling page and not into a common function (this is the not common part).
So i don't understand the goal of this PR. It moves all the "non common" part of code into a common function forcing then to make switch to execute the non common code.
???

Reply:
The are a few ideas behind this PR
1. Reduce the amount of the parameter that are need for the `showDocuments` function
2. Reduce the dobule of code lines.
3. And the goal in the long run is to remove the different behavior of pages, when possible, like
`$urlsource = $_SERVER['PHP_SELF'].'?id='.$object->id;`
and
`$urlsource = $_SERVER['PHP_SELF'].'?factid='.$object->id;`